src/foam/u2/dialog/Popup.js

```diff
@@ -150,13 +159,12 @@ foam.CLASS({
     function open() {
       this.document.body.insertAdjacentHTML('beforeend', this.outerHTML);
       this.load();
-    }
-  ],
+    },
 
-  listeners: [
-    function close() {
-      if ( this.onClose ) this.onClose();
-      this.remove();
+    function close(closingModal) {
```

Comment:
Why do we need a promise?

Reply:
When calling remove() inside close it is detaching the view and all the registered listeners were cleared before the pub call. Promise allows the pub call to hit first before detaching.